lib/osutil/traversessymlink.go

```diff
@@ -35,19 +35,21 @@ func (e NotADirectoryError) Error() string {
 // TraversesSymlink returns an error if any path component of name (including name
 // itself) traverses a symlink.
 func TraversesSymlink(filesystem fs.Filesystem, name string) error {
-	var err error
-	name, err = fs.Canonicalize(name)
-	if err != nil {
-		return err
-	}
+	return TraversesSymlinkSegment(filesystem, name, ".")
+}
 
+// TraversesSymlinkSegment returns an error if any path component of name below
+// parent (including name itself) traverses a symlink. Expects
+// `fs.IsParent(name, parent) == true`, otherwise behaviour is undefined.
+func TraversesSymlinkSegment(filesystem fs.Filesystem, name, parent string) error {
```

Comment:
This IsParent requirement doesn't sound like it would hold for the dot we're passing in, i.e. `IsParent("whatever", ".") => false` I would expect? Is `filepath.Rel` cool with this?
 
(Also out parameter order in IsParent is the opposite of filepath.Rel which is at the very least confusing... But not your fault or a thing in this PR :)

Reply:
`IsParent` is aware that "." is special, there's even a test case confirming it.

Reply:
Make a note of that in the comment please, otherwise it does look like we're saying you cannot do what we specifically do a couple of lines higher up.